core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -401,6 +434,39 @@ private static void handleException(ClientRequestContext ctx, HttpRequestDuplica
         };
     }
 
+    private void redirect(ClientRequestContext ctx, ClientRequestContext derivedCtx,
+                          HttpRequestDuplicator rootReqDuplicator, CompletableFuture<HttpResponse> future,
+                          CompletableFuture<Void> returnedResWhenComplete, HttpResponse originalRes,
+                          Backoff backoff, RequestHeaders newHeaders) {
+        final HttpRequest originalReq = getOriginalRequest(ctx);
+        assert originalReq != null;
+        final String originalPath = originalReq.path();
+        final String newPath = newHeaders.path();
+        if (newPath.equals(originalPath) || !addRedirectPath(ctx, newPath)) {
+            final Set<String> paths = redirectPaths(ctx);
+            assert paths != null;
+            final RedirectLoopsException exception = new RedirectLoopsException(originalPath, paths);
+            abortResponse(originalRes, derivedCtx, exception);
+            handleException(ctx, rootReqDuplicator, future, exception, false);
+            return;
+        }
+        abortResponse(originalRes, derivedCtx, null);
+        resetTotalAttempts(ctx);
```

Comment:
Don't we need to limit the maximum number of redirections? https://www.websense.com/content/support/library/web/hosted/fw_redirect/fwr_browser_redirect_limit.aspx

Reply:
Because we are checking `RedirectLoopsException`, I thought it's okay to reset on every redirection. Do you have a concern?